beacon-chain/operations/synccommittee/signature.go

```diff
@@ -0,0 +1,49 @@
+package synccommittee
+
+import (
+	types "github.com/prysmaticlabs/eth2-types"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/copyutil"
+)
+
+// SaveSyncCommitteeSignature saves a sync committee signature in cache.
+// The cache does not filter out duplicate signature, it will be up to the caller.
+func (s *Store) SaveSyncCommitteeSignature(sig *ethpb.SyncCommitteeSignature) error {
```

Comment:
If we had generics...
This code seems like a full copy of logic from the contribution code

Reply:
Yep, once we have generics, other objects like `attestation`, `slashing`... etc can all be consolidate under one